target/linux/ipq40xx/base-files/lib/preinit/05_set_iface_mac_ipq40xx.sh

```diff
@@ -9,6 +9,13 @@ preinit_set_mac_address() {
 		ip link set dev eth0 address $(macaddr_add "$base_mac" +1)
 		ip link set dev eth1 address $(macaddr_add "$base_mac" +3)
 		;;
+	xiaomi,meshd01)
+		base_mac=$(mtd_get_mac_binary "0:ART" 0x0)
+		ip link set dev eth0 address "${base_mac}"
+		ip link set dev eth1 address $(macaddr_add "${base_mac}" 1)
+		ip link set dev eth2 address $(macaddr_add "${base_mac}" 2)
+		ip link set dev eth3 address $(macaddr_add "${base_mac}" 3)
```

Comment:
Have you checked whether there are additional addresses in "0:ART", e.g. at 0x6 etc.?
If yes, please use them directly instead of doing arithmetics.

Reply:
just checked, yes, it have, will fix it